cachi2/core/models/sbom.py

```diff
@@ -93,3 +96,173 @@ class Sbom(pydantic.BaseModel):
+                        )
+                        for prop in component.properties
+                    ],
+                )
+                for component in self.components
+            ],
+            creationInfo=SPDXCreationInfo(
+                creators=[f"{tool.vendor} {tool.name}" for tool in self.metadata.tools]
+            ),
+        )
+
+
+class SPDXPackageExternalRef(pydantic.BaseModel):
+    """SPDX Package External Reference.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/#721-external-reference-field
+    """
+
+    referenceCategory: str = pydantic.Field(
+        pattern="^SECURITY|PACKAGE-MANAGER|PERSISTENT-ID|OTHER$"
+    )
+    referenceLocator: str
+    referenceType: str
+
+    @pydantic.model_validator(mode="after")
+    def _model_validator(self: Self) -> Self:
+        if self.referenceCategory == "SECURITY":
+            if self.referenceType not in ["cpe22", "cpe23", "advisory", "fix", "url", "swid"]:
+                raise ValueError("Invalid reference type for SECURITY category")
+        elif self.referenceCategory == "PACKAGE-MANAGER":
+            if self.referenceType not in ["maven-central", "npm", "nuget", "bower", "purl"]:
+                raise ValueError("Invalid reference type for PACKAGE-MANAGER category")
+        elif self.referenceCategory == "PERSISTENT-ID":
+            if self.referenceType not in ["swh", "gitbom"]:
+                raise ValueError("Invalid reference type for PERSISTENT-ID category")
+        return self
+
+    def __hash__(self) -> int:
+        return hash((self.referenceLocator, self.referenceType, self.referenceCategory))
+
+
+class SPDXPackageAnnotation(pydantic.BaseModel):
+    """SPDX Package Annotation.
+
+    Compliant to the SPDX specification:
+    https://github.com/spdx/spdx-spec/blob/development/v2.3/schemas/spdx-schema.json#L237
+    """
+
+    annotator: str
+    annotationDate: str
+    annotationType: Literal["OTHER", "REVIEW"]
+    comment: str
+
+    def __hash__(self) -> int:
+        return hash((self.annotator, self.annotationDate, self.annotationType, self.comment))
+
+
+class SPDXPackage(pydantic.BaseModel):
+    """SPDX Package.
+
+    Compliant to the SPDX specification:
+    https://spdx.github.io/spdx-spec/v2.3/package-information/
+    """
+
+    name: str
+    versionInfo: Optional[str] = None
+    externalRefs: list[SPDXPackageExternalRef] = []
+    annotations: list[SPDXPackageAnnotation] = []
+
+    @classmethod
+    def from_package_dict(cls, package: dict[str, Any]) -> "SPDXPackage":
+        """Create a SPDXPackage from a Cachi2 package dictionary."""
+        external_refs = []
+        for er in package.get("externalRefs", []):
+            external_refs.append(SPDXPackageExternalRef(**er))
+        annotations = []
+        for an in package.get("annotations", []):
+            annotations.append(SPDXPackageAnnotation(**an))
```

Comment:
This looks almost perfect! I wonder (don't insist, just wonder) why not to simply declare what annotations are:
```suggestion
annotations = [SPDXPackageAnnotation(**a) for a in package.get("annotations", [])]
```
instead of providing an instruction of how to populate the list?